identity-account/src/types/signature.rs

```diff
@@ -1,10 +1,10 @@
-// Copyright 2020-2021 IOTA Stiftung
+// Copyright 2020-2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
 use identity_core::crypto::PublicKey;
 
 /// A digital signature and associated public key.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Deserialize)]
 pub struct Signature {
   pub(crate) pkey: PublicKey,
   pub(crate) data: Vec<u8>,
```

Comment:
I think I commented about this on the other PR but do we need `Signature` to contain the `PublicKey` as well? I don't think it's used. Is it needed for future signature schemes?

Not related to the changes in this PR but it is about the `Storage` implementation.

Reply:
No we don't need it. In fact, this is unused in the account, too. The account `Signature` type could need a refactoring to simply return either a `Vec<u8>` or a newtype wrapper around it. The `pkey` is simply [discarded in `RemoteSign`](https://github.com/iotaledger/identity.rs/blob/267574cd0a3a4edfaa7876ccdae7151d9de1e6b6/identity-account/src/crypto/remote.rs#L94). Could be part of a number of storage refactors mentioned before (redesign errors, remove `set_password`, refactor `Signature`)?